src/test/java/org/eclipse/lyo/validation/ValidationExample.java

```diff
@@ -33,30 +43,74 @@
     /**
      * Loads an example SHACL file and validates all focus nodes against all shapes.
      */
-    public static void main(String[] args) throws Exception {
+	public static void validateModelAgainstShaclModel() throws IllegalAccessException, InvocationTargetException, DatatypeConfigurationException, OslcCoreApplicationException {
+		
+		// Load the main data model
+		Model dataModel = ModelFactory.createDefaultModel();
+		dataModel.read(ValidationExample.class.getResourceAsStream("/aResource-Data.ttl"), "urn:dummy", FileUtils.langTurtle);
 
-        // Load the main data model
-        Model dataModel = ModelFactory.createDefaultModel();
-        dataModel.read(ValidationExample.class.getResourceAsStream("/aResource-Data.ttl"),
-                "urn:dummy", FileUtils.langTurtle);
-        dataModel.write(System.out, "TURTLE");
+		// Load the SHACL shape model
+		Model shapeModel = ModelFactory.createDefaultModel();
+		shapeModel.read(ValidationExample.class.getResourceAsStream("/aResource-Shape.ttl"), "urn:dummy", FileUtils.langTurtle);
+
+		Validator validator = new ValidatorImpl();
+        //Validate the data model against the SHACL shapes model. You can have shape and data in same model.
+		ValidationResultModel validationResult = validator.validate(dataModel, shapeModel);
+        for (ResourceModel rm : validationResult.getInvalidResources()) {
+            if (!rm.getResult().isValid()) {
+                System.out.println(rm.getResult().show());
+            } else {
+                System.out.println("Resource is Valid");
```

Comment:
you are looping over `getInvalidResources()`, are you sure this line will ever get executed? if it will, there is a serious API design issue here.

Reply:
well, this is where I need @yashkhatri on this review. Does this mean @yashkhatri cannot even contribute to this project?